src/doc/doc_phases/doc_graphs.nit

```diff
@@ -0,0 +1,123 @@
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Adds importation and class hierarchy graphs.
+module doc_graphs
+
+import doc_poset
+
+redef class ToolContext
+
+	# Do not generate `graphviz` diagrams.
+	var opt_nodot = new OptionBool("do not generate graphes with graphviz", "--no-dot")
+
+	redef init do
+		super
+		option_context.add_option(opt_nodot)
+	end
+end
+
+# This phase insert importation and inheritance graphs into pages.
+class GraphPhase
+	super DocPhase
+
+	redef fun apply(doc) do
+		if ctx.opt_nodot.value then return
+		for page in doc.pages do
+			if page isa MEntityPage then
+				var article = page.build_graph(self, doc)
+				if article != null then
+					# FIXME avoid diff
+					# page.root.add article
+					page.root.children[1].children.insert(article, 0)
```

Comment:
trop d'identation

~~~
for page in doc.pages do
	var article = page.build_graph(self, doc)
	if article == null then continue
	# FIXME avoid diff
	# page.root.add article
	page.root.children[1].children.insert(article, 0)
end
~~~